coral/src/app/features/topics/schema-request/components/TopicSchema.tsx

```diff
@@ -0,0 +1,151 @@
+import { Controller as _Controller, useFormContext } from "react-hook-form";
+import { ChangeEvent, useEffect, useState } from "react";
+import { BorderBox, Box, Typography } from "@aivenio/aquarium";
+import MonacoEditor from "@monaco-editor/react";
+import { editor } from "monaco-editor";
+import { readFile } from "src/app/features/topics/schema-request/utils/read-file";
+import { TopicRequestFormSchema } from "src/app/features/topics/schema-request/utils/zod-schema";
+import { FileInput } from "src/app/components/FileInput";
+import isString from "lodash/isString";
+
+type TopicSchemaProps = {
+  name: keyof TopicRequestFormSchema;
+  required: boolean;
+};
+
+function TopicSchema(props: TopicSchemaProps) {
+  const { name, required } = props;
+  const { setValue, setError, clearErrors, control } =
+    useFormContext<TopicRequestFormSchema>();
+
+  const [schema, setSchema] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
+    if (isString(schema)) {
+      setValue(name, schema, {
+        shouldValidate: true,
+        shouldTouch: true,
+        shouldDirty: true,
+      });
+    }
+  }, [schema]);
+
+  function checkEmptyFile(event: ChangeEvent<HTMLInputElement>) {
+    if (!required) return;
+    const file = event.target?.files?.[0];
+    if (!file) {
+      setError(name, { message: "File is a required field" });
+    }
+  }
+
+  function validateSchema(markers: editor.IMarker[]) {
+    if (markers.length > 0) {
+      setError(name, { message: markers?.[0]?.message, type: "custom" });
+      setValue(name, undefined, {
+        shouldValidate: false,
+        shouldDirty: true,
+        shouldTouch: true,
+      });
+    }
+  }
+
+  function uploadFile(event: ChangeEvent<HTMLInputElement>) {
+    const file = event.target?.files?.[0];
+    if (file) {
+      // eslint-disable-next-line no-inner-declarations
+      async function getFile() {
```

Comment:
Why we declare this function? Wont the following work?

```
readFile(file).then((fileContent) => {
  clearErrors();
  setSchema(fileContent);
});
```

Reply:
Good catch, I don't need that anymore! 👯  That's a leftover from me trying to do that without a local state prop, so make the effect hood block until the file is read. Will remove it!